bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,254 @@
+            updateStatus(ThingStatus.OFFLINE);
+        }
+        return null;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(VektivaSmarwiConfiguration.class);
+        logger.debug("IP address: {}", config.ip);
+
+        future = scheduler.scheduleWithFixedDelay(this::checkStatus, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private synchronized void initializeWebSocketSession() {
+        if (config.useWebSockets) {
+            closeSession();
+            session = createSession();
+            if (session != null) {
+                logger.debug("WebSocket connected!");
+            }
+        }
+    }
+
+    private void checkStatus() {
+        String url = "http://" + config.ip + "/statusn";
+
+        try {
+            ContentResponse resp = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            logger.debug("status values: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                processStatusResponse(resp.getContentAsString());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "got response code: " + resp.getStatus());
+            }
+            // reconnect web socket if not connected
+            if (config.useWebSockets && (session == null || !session.isOpen())
+                    && ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                logger.debug("Initializing WebSocket session");
+                initializeWebSocketSession();
+            }
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "exception during status getting");
+            logger.debug("Exception during status update", e);
+            session = null;
+        }
+    }
+
+    public synchronized void processStatusResponse(String content) {
+        if (!ThingStatus.ONLINE.equals(getThing().getStatus())) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        String[] values = content.split("\n");
+
+        updateProperty("type", getPropertyValue(values, "t"));
+        updateProperty(Thing.PROPERTY_FIRMWARE_VERSION, getPropertyValue(values, "fw"));
+        updateProperty("rssi", getPropertyValue(values, "rssi"));
+        updateProperty("name", getPropertyValue(values, "cid"));
+        updateProperty("status", getPropertyValue(values, "s"));
+        updateProperty("error", getPropertyValue(values, "e"));
+        updateProperty("ok", getPropertyValue(values, "ok"));
+        updateProperty("ro", getPropertyValue(values, "ro"));
+        updateProperty("fix", getPropertyValue(values, "fix"));
+
+        if ("1".equals(getPropertyValue(values, "ro"))) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, "Online but not ready!");
+        }
+
+        int position = getPropertyValue(values, "pos").equals("o") ? 0 : 100;
```

Comment:
Just checking: Is the value of `"pos"` the character `"o"` or the number zero `"0"`? Because the code compares with a character `"o"`?

Reply:
yeah it means o-pen position of the window